src/extras/global.js

```diff
@@ -10,18 +10,25 @@
   var firstGlobalProp, secondGlobalProp, lastGlobalProp;
   function getGlobalProp () {
     var cnt = 0;
-    var lastProp;
+    var foundLastProp, result;
     for (var p in global) {
       // do not check frames cause it could be removed during import
       if (shouldSkipProperty(p))
         continue;
       if (cnt === 0 && p !== firstGlobalProp || cnt === 1 && p !== secondGlobalProp)
         return p;
+      if (systemJSPrototype.firstGlobalProp) {
+        if (foundLastProp) {
+          lastGlobalProp = p;
```

Comment:
This is a good catch with the race issue! Perhaps it should also apply to both branches as well?